presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1337,10 +1337,10 @@ private void analyzeHaving(QuerySpecification node, Scope scope)
                 Expression expression = item.getSortKey();
 
                 Expression orderByExpression = null;
-                if (expression instanceof QualifiedNameReference && !((QualifiedNameReference) expression).getName().getPrefix().isPresent()) {
+                if (expression instanceof Identifier) {
                     // if this is a simple name reference, try to resolve against output columns
 
-                    QualifiedName name = ((QualifiedNameReference) expression).getName();
+                    QualifiedName name = QualifiedName.of(((Identifier) expression).getName());
```

Comment:
You could wrap this expression as function:

```
QualifiedName.from(expression)
```

Somethig similar to `SymbolReference#from`

That way it will be less boilerplate